src/overlays/actors/ovl_En_Pst/z_en_pst.c

```diff
@@ -31,51 +138,312 @@ const ActorInit En_Pst_InitVars = {
+                return D_80B2C448;
+            case POSTBOX_SOUTH_LOWER_CLOCKTOWN:
+                return D_80B2C468;
+            default:
+                return NULL;
+        }
+    } else if (this->stateFlags & 0x20) {
+        if (this->exchangeItemId == EXCH_ITEM_LETTER_MAMA) {
+            return D_80B2C488;
+        } else {
+            return D_80B2C490;
+        }
+    } else {
+        this->msgEventCallback = EnPst_ChooseBehaviour;
+        switch (this->actor.params) {
+            case POSTBOX_SOUTH_UPPER_CLOCKTOWN:
+                return D_80B2C23C;
+            case POSTBOX_NORTH_CLOCKTOWN:
+                return D_80B2C288;
+            case POSTBOX_EAST_UPPER_CLOCKTOWN:
+                return D_80B2C2D4;
+            case POSTBOX_EAST_LOWER_CLOCKTOWN:
+                return D_80B2C320;
+            case POSTBOX_SOUTH_LOWER_CLOCKTOWN:
+                return D_80B2C36C;
+            default:
+                return NULL;
+        }
+    }
+}
+
+s32 EnPst_CheckTalk(EnPst* this, PlayState* play) {
+    Player* player = GET_PLAYER(play);
+    s32 ret = false;
+
+    if (this->stateFlags & 7) {
+        if (Actor_ProcessTalkRequest(&this->actor, &play->state)) {
+            this->stateFlags &= ~0x30;
+            if (player->exchangeItemId == EXCH_ITEM_LETTER_TO_KAFEI) {
+                this->stateFlags |= 0x10;
+                this->exchangeItemId = player->exchangeItemId;
+            } else if (player->exchangeItemId != EXCH_ITEM_NONE) {
+                this->stateFlags |= 0x20;
+                this->exchangeItemId = player->exchangeItemId;
+            }
+            // If Letter to Kafei is deposited, value is set to 2
+            this->isLetterToKafeiDeposited = EnPst_HandleLetterDay1(this);
+            SubS_UpdateFlags(&this->stateFlags, 0, 7);
+            this->behaviour = 0;
+            this->msgEventCallback = NULL;
+            this->stateFlags |= 0x40;
+            this->msgEventScript = EnPst_GetMsgEventScript(this, play);
+            this->actionFunc = EnPst_Talk;
+            ret = true;
+        }
+    }
+    return ret;
+}
+
+s32 EnPst_UpdateFlagsSubs(EnPst* this, PlayState* play, ScheduleOutput* scheduleOutput) {
+    SubS_UpdateFlags(&this->stateFlags, 3, 7);
+    return true;
+}
 
-#endif
+s32 EnPst_ProcessScheduleOutput(EnPst* this, PlayState* play, ScheduleOutput* scheduleOutput) {
+    s32 ret = false;
 
-extern ColliderCylinderInit D_80B2C4B8;
-extern CollisionCheckInfoInit2 D_80B2C4E4;
+    this->stateFlags = 0;
 
-extern UNK_TYPE D_06001A80;
+    switch (scheduleOutput->result) {
+        case POSTBOX_SCH_AVAILABLE:
+            ret = EnPst_UpdateFlagsSubs(this, play, scheduleOutput);
+            break;
+        case POSTBOX_SCH_CHECKED_BY_POSTMAN:
+            ret = true;
+            break;
+    }
+    return ret;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Pst/func_80B2B830.s")
+void EnPst_HandleSchedule(EnPst* this, PlayState* play) {
+}
```

Comment:
? This is a do nothing function?

is this named "HandleSchedule" as some boilerplate requirement of the schedule system?

Reply:
The thing is this function is never set as the action function, so it isn't really the DoNothing func. Instead it does look like one of the functions that while not needed, actors that use the Schedule system seem to have.

Reply:
There should be a comment explaining the function's naming choice, as it clearly does nothing and does not handle any schedule.

Reply:
(At least for me, I think DoNothing should be relegated to just actionFuncs to keep empty actor action funcs separate from other types of funcs. For empty non-action funcs I'd prefer something like NoOp or Empty if the context doesn't give a better name.)

I'd agree with that. A comment explaining the context would be beneficial.